src/io/s3_filesys.cc

```diff
@@ -492,33 +769,47 @@ class ReadStream : public CURLReadStreamBase {
 void ReadStream::InitRequest(size_t begin_bytes,
                              CURL *ecurl,
                              curl_slist **slist) {
-  // initialize the curl request
-  std::vector<std::string> amz;
+  std::string payload = "";
+  time_t curr_time = time(NULL);
```

Comment:
you don’t ever need to assign to an empty string. in fact, it adds unnecessary overhead